apps/speed.c

```diff
@@ -1610,6 +1646,17 @@ int speed_main(int argc, char **argv)
             }
             doit[D_EVP_HMAC] = 1;
             break;
+        case OPT_CMAC:
+#ifndef OPENSSL_NO_CMAC
+            evp_cmac_cipher = EVP_get_cipherbyname(opt_arg());
+            if (evp_cmac_cipher == NULL) {
+                BIO_printf(bio_err, "%s: %s is an unknown cipher\n",
+                           prog, opt_arg());
+                goto end;
+            }
+            doit[D_EVP_CMAC] = 1;
+#endif
+            break;
```

Comment:
Would an error if this option is used when not supported be better?
At least the argument should be consumed.

Reply:
currently, the if not supported the option is not known at all (will print error: "Use -help for summary."
similar to NO_FORK, OPENSSL_NO_ASYNC and OPENSSL_NO_ENGINE).
But I can print an error here, and ignore the argument.

Reply:
Do what's consistent which seems to be not knowing about the option if it isn't included :(
Moving the `#ifndef` up one line and the `#endif` down one achieves this.
I don't like it but that's the standard here.

Pauli

Reply:
It would regardless since the option isn't defined.
Moving the preprocessor directives would be consistent with the rest of the code but I'm not fussed.

Reply:
the switch has no default, I was not sure if it would result in a warning.